### PR TITLE
Bloomfilter intersect/decrement/typification

### PR DESCRIPTION
**Pull request by 0xxon**
This PR adds some new bloomfilter functionality - mostly intersect and decrement - and fixes a typification bug.

### CODE REVIEW

**Comment by 0xxon** [comment]
This fixes #1938

File: src/OpaqueVal.cc [hunk 1]
```diff
@@ -701,6 +701,49 @@ BloomFilterValPtr BloomFilterVal::Merge(const BloomFilterVal* x, const BloomFilt
...
+BloomFilterValPtr BloomFilterVal::Intersect(const BloomFilterVal* x, const BloomFilterVal* y)
+	{
+	if ( x->Type() && // any one 0 is ok here
+	     y->Type() && ! same_type(x->Type(), y->Type()) )
+		{
+		reporter->Error("cannot merge Bloom filters with different types");
+		return nullptr;
+		}
+
+	auto final_type = x->Type() ? x->Type() : y->Type();
```
Comment (timwoj) [inline src/OpaqueVal.cc:725]: This could move down closer to where it gets used (line 735?)

File: src/probabilistic/bloom-filter.bif [hunk 2]
```diff
@@ -160,6 +160,9 @@ function bloomfilter_add%(bf: opaque of bloomfilter, x: any%): any
 
 ## Decrements the counter for an element that was added to a counting bloom filter in the past.
 ##
+## Note that decrement operations can use to false negatives if used on a counting bloom-filter
```
Comment (timwoj) [inline src/probabilistic/bloom-filter.bif:163]: s/can use/can lead/

File: src/OpaqueVal.cc [hunk 3]
```diff
@@ -701,6 +701,49 @@ BloomFilterValPtr BloomFilterVal::Merge(const BloomFilterVal* x, const BloomFilt
...
+
+	if ( typeid(*x->bloom_filter) != typeid(*y->bloom_filter) )
+		{
+		reporter->Error("cannot merge different Bloom filter types");
+		return nullptr;
+		}
+
+	if ( typeid(*x->bloom_filter) != typeid(*y->bloom_filter) )
```
[the remark this answers is not in the document]
Comment (0xxon) [reply src/OpaqueVal.cc:733]: Heh, forgot to delete the first one. Thank you.

**Review by timwoj — APPROVED** [review]

**Pull request merged**